### PR TITLE
Update GOVERNANCE.md to better allow non-code maintainers

### PR DESCRIPTION
**Pull request by youngnick**
Updates #20.

Also includes some other wording updates to modernize this document.

Signed-off-by: Nick Young <ynick@vmware.com>

### CODE REVIEW

File: GOVERNANCE.md [hunk 1]
```diff
@@ -8,20 +8,27 @@ This document defines the project governance for Contour.
...
+* **Contributors:** Regular contributions to projects (documentation, code reviews, responding to issues, participation in proposal discussions, contributing code, etc.). For the Contour project and all its associated repositories, becoming a *Contributor* means becoming a *Github Organization Member*, with the ability to label issues and PRs. 
+* **Maintainers**: The Contour project leaders.
+They are responsible for the overall health and direction of the project; final reviewers of PRs and responsible for releases.
+Some Maintainers are responsible for one or more components within a project, acting as technical leads for that component.
+Maintainers are expected to do some or all of the below:
+  - contribute code and/or documentation, review PRs including ensuring quality of code, triage issues, proactively fix bugs, and perform maintenance tasks for these components,
+  - or perform other equivalent non-code contributions, such as managing the Contour community, promoting and marketing Contour, encouraging new contributors, or helping promote contributors to become maintainers.
```
Comment (stevesloka) [inline GOVERNANCE.md:31]: How does being a maintainer help in fulfilling these roles?

File: GOVERNANCE.md [hunk 2]
```diff
@@ -8,20 +8,27 @@ This document defines the project governance for Contour.
...
+All the code repositories owned by the Project Contour Github Organization are
+governed by Contour community and maintained under the `ProjectContour` namespace.
+The following repos are the most noteworthy:
 
 * **[Contour](https://github.com/ProjectContour/Contour):** Main Contour codebase.
-* **[Gimbal](https://github.com/ProjectContour/Gimbal):** Gimbal is an ingress load balancing platform capable of routing traffic to multiple Kubernetes and OpenStack clusters
 * **[community](https://github.com/ProjectContour/community):** Used to store community-related material–e.g., proposals, presentation slides, governance documents, community meeting minutes, etc.
+* **[contour-operator](https://github.com/ProjectContour/Contour):** The Contour
+Operator is a tool for installing and managing Contour.
 * **[ir2proxy ](https://github.com/ProjectContour/ir2proxy):** ir2proxy is a tool to convert Contour's IngressRoute resources to HTTPProxy resources.
```
Comment (stevesloka) [inline GOVERNANCE.md:19]: Ingress route has been deprecated for some time, might as well remove it as well.

File: GOVERNANCE.md [hunk 3]
```diff
@@ -8,20 +8,27 @@ This document defines the project governance for Contour.
...
-* **Maintainers**: The Contour project leaders. They are responsible for the overall health and direction of the project; final reviewers of PRs and responsible for releases. Some Maintainers are responsible for one or more components within a project, acting as technical leads for that component. Maintainers are expected to contribute code and documentation, review PRs including ensuring quality of code, triage issues, proactively fix bugs, and perform maintenance tasks for these components.
+* **Contributors:** Regular contributions to projects (documentation, code reviews, responding to issues, participation in proposal discussions, contributing code, etc.). For the Contour project and all its associated repositories, becoming a *Contributor* means becoming a *Github Organization Member*, with the ability to label issues and PRs. 
+* **Maintainers**: The Contour project leaders.
+They are responsible for the overall health and direction of the project; final reviewers of PRs and responsible for releases.
+Some Maintainers are responsible for one or more components within a project, acting as technical leads for that component.
```
Comment (stevesloka) [inline GOVERNANCE.md:28]: What defines this role and who is responsible for each? We're all technical leads for all components given it's such a small team.

File: GOVERNANCE.md [hunk 4]
```diff
@@ -8,20 +8,27 @@ This document defines the project governance for Contour.
...
-* **[Gimbal](https://github.com/ProjectContour/Gimbal):** Gimbal is an ingress load balancing platform capable of routing traffic to multiple Kubernetes and OpenStack clusters
 * **[community](https://github.com/ProjectContour/community):** Used to store community-related material–e.g., proposals, presentation slides, governance documents, community meeting minutes, etc.
+* **[contour-operator](https://github.com/ProjectContour/Contour):** The Contour
+Operator is a tool for installing and managing Contour.
 * **[ir2proxy ](https://github.com/ProjectContour/ir2proxy):** ir2proxy is a tool to convert Contour's IngressRoute resources to HTTPProxy resources.
 
 
 ## Community Roles
-
+All of the below roles share the responsibility to do their best to
```
[the remark this answers is not in the document]
Comment (youngnick) [reply GOVERNANCE.md:23]: 🤦 thanks.

File: GOVERNANCE.md (hunk 1 above)
Comment (youngnick) [reply GOVERNANCE.md:31]: Being a maintainer is our highest level of contribution. Having a way in which non-code contributions can lead to you being a maintainer is a way to encourage non-code contributors, and ensure that there's a path for non-code contributors to have a strong say in the project's direction.

File: GOVERNANCE.md (hunk 3 above)
Comment (youngnick) [reply GOVERNANCE.md:28]: This is in there to cover us for if and when we scale the team and have more maintainers. Ideally, for example, we end up with maintainers who are responsible for the docs and website. This covers that work as well.

Basically this just says "To be a maintainer, you don't have to be knowledgeable or responsible across the entire project".

**Review by tsaarni — APPROVED** [review]

File: GOVERNANCE.md (hunk 3 above)
Comment (stevesloka) [reply GOVERNANCE.md:25]: I think the names are holding us back in addition to how they are used when interacting with the CNCF. Kubernetes has an excellent page on this which outlines the roles + responsibilities of each: https://github.com/kubernetes/community/blob/master/community-membership.md

File: GOVERNANCE.md (hunk 1 above)
Comment (stevesloka) [reply GOVERNANCE.md:28]: It still feels too vague to me. There are lots of folks who contribute to Contour that fit into what's described here. I mentioned this below, but I think defining the roles and responsibilities makes it clear what folks are expected to do at each level of each role. (ref: https://github.com/projectcontour/community/pull/23#discussion_r748542731)

**Comment by youngnick** [comment]
Thanks for the comments @stevesloka. I think we should discuss this one at the maintainer's meeting tomorrow. I'd ask that you have a think about what sort of governance you would like - how many levels should there be? What is the process to change levels? And so on.

The idea is not to make Contour maintainership hard, it's to make it easier to become a maintainer, so we can get more people interested in it.

File: GOVERNANCE.md (hunk 1 above)
Comment (stevesloka) [reply GOVERNANCE.md:28]: > managing the Contour community, promoting and marketing Contour, encouraging new contributors, or helping promote contributors to become maintainers.

This feels like an Ambassador role not a maintainer. I could go give talks on GatewayAPI, write up blog posts, point people to the repo, but wouldn't become a maintainer doing that work. That work is important to any project, but does that fit into the role of "maintainer"? Doesn't feel 100% like that to me.

**Comment by stevesloka** [comment]
> Thanks for the comments @stevesloka. I think we should discuss this one at the maintainer's meeting tomorrow. I'd ask that you have a think about what sort of governance you would like - how many levels should there be? What is the process to change levels? And so on.

https://github.com/projectcontour/community/pull/23#discussion_r748542731

> The idea is not to make Contour maintainership hard, it's to make it easier to become a maintainer, so we can get more people interested in it.

Yup I agree, but you need a way to get people interested first which has been the struggle. We need to be consistent with folks who do contribute regularly and to get them moved up into the proper roles. Here's a good thread on this recently: https://twitter.com/rakyll/status/1459623476201881602

**Comment by youngnick** [comment]
So, it seems like you are saying that you have some guidelines you would like to see., that will have us end up with a more role-based structure. That's great! I was kind of trying to move with a more minimal strategy for this one, but I agree that what you're suggesting is much better.

Should I close this PR so you can open your one instead?

**Review by sunjayBhatia — COMMENTED** [review]
Not to complicate this more, but some thoughts:

- looking at prior art on this (envoy, Istio, k8s docs that Steve linked, etc.) it seems 'Maintainer' is primarily a technical role, focused on code contribution, technical responsibilities, catchall project health etc. with a specific path to achieve that status
  - just for consistency and simplicity, it might make sense to keep it this way
- Istio has a nice breakdown of different roles that we could crib from: https://github.com/istio/community/blob/master/ROLES.md#triager
  - similar to k8s roles, they have descriptions and some explicit paths on how to achieve that role
  - particularly the "Documentation Reviewer" role that we wanted to promote to encourage contributions in this area, related to the initiative Jonas started
  - particularly the "Triager" role that we could massage a bit to be separate from a "Maintainer" and encompass more of a traditional "PM" role, someone who has technical background and helps set project priority etc., doesn't necessarily have to also make code contributions
- we could also specifically add a "Community Manager" or some similar name role and explicitly lay out the responsibilities of that role and how they are appointed, just to be explicit  how that role functions in the project, for external visibility etc.

**Comment by skriss** [comment]
I agree with the sentiment of defining some additional roles with specific responsibilities, rather than trying to shoehorn everything into a single "Maintainer" role. I think this would make expectations more clear, both for those looking to join the team, as well as for the existing team to make decisions about prospective new members. As a starting point I'd definitely be in favor of defining a "Community Manager" role and a "Docs Reviewer" or "Docs Maintainer"  role.

**Comment by youngnick** [comment]
Thanks for the feedback everyone, I'll write up something a bit more comprehensive.

**Comment by tsaarni** [comment]
I think fine grained roles would typically be defined also to facilitate security, that is, to make it possible to give every role only the minimum required access to the systems that the project has.

**Comment by jonasrosland** [comment]
After discussion with @caniszczyk the recommendation is to create something like "Doc Maintainers" or "SIG Docs" or "SIG Community Management" within our MAINTAINERS.md, that way the maintainers description can keep as is and we have more flexibility when we want to add more responsibility to awesome contributors.

**Comment by jonasrosland** [comment]
Example PR from Harbor: https://github.com/goharbor/community/pull/180

**Comment by xaleeks** [comment]
I like 'SIG Community Management'

**Comment by youngnick** [comment]
Yes, if we're planning for growth, establishing SIGs seems like a good idea.

**Comment by youngnick** [comment]
This has been superseded by #25. Closing.

**Pull request closed without merging**